src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -4126,5 +4127,1202 @@ static void Main()
+            VerifyParameterSymbol(method.Parameters[0], "scoped R<System.Int32> x", RefKind.None, DeclarationScope.ValueScoped);
+            VerifyParameterSymbol(method.Parameters[1], "scoped in System.Int32 y", RefKind.In, DeclarationScope.RefScoped);
+        }
+
+        [Fact]
+        public void RetargetingParameter()
+        {
+            var sourceA =
+@"public ref struct R { }
+public class A
+{
+    public static void F(scoped R x, scoped in int y) { }
+}
+";
+            var comp = CreateCompilation(sourceA, targetFramework: TargetFramework.Mscorlib40);
+            var refA = comp.ToMetadataReference();
+
+            var sourceB =
+@"class B
+{
+    static void Main()
+    {
+        A.F(default, 0);
+    }
+}";
+            comp = CreateCompilation(sourceB, new[] { refA }, targetFramework: TargetFramework.Mscorlib45);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp);
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var expr = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single().Expression;
+            var method = model.GetSymbolInfo(expr).Symbol.GetSymbol<RetargetingMethodSymbol>();
+
+            VerifyParameterSymbol(method.Parameters[0], "scoped R x", RefKind.None, DeclarationScope.ValueScoped);
+            VerifyParameterSymbol(method.Parameters[1], "scoped in System.Int32 y", RefKind.In, DeclarationScope.RefScoped);
+        }
+
+        private static readonly SymbolDisplayFormat displayFormatWithScoped = SymbolDisplayFormat.TestFormat.
+            WithCompilerInternalOptions(SymbolDisplayCompilerInternalOptions.IncludeScoped).
+            AddLocalOptions(SymbolDisplayLocalOptions.IncludeRef);
+
+        private static void VerifyParameterSymbol(ParameterSymbol parameter, string expectedDisplayString, RefKind expectedRefKind, DeclarationScope expectedScope)
+        {
+            Assert.Equal(expectedRefKind, parameter.RefKind);
+            Assert.Equal(expectedScope, parameter.Scope);
+            Assert.Equal(expectedDisplayString, parameter.ToDisplayString(displayFormatWithScoped));
+
+            var attribute = parameter.GetAttributes().FirstOrDefault(a => a.GetTargetAttributeSignatureIndex(parameter, AttributeDescription.LifetimeAnnotationAttribute) != -1);
+            Assert.Null(attribute);
+
+            VerifyParameterSymbol(parameter.GetPublicSymbol(), expectedDisplayString, expectedRefKind, expectedScope);
+        }
+
+        private static void VerifyParameterSymbol(IParameterSymbol parameter, string expectedDisplayString, RefKind expectedRefKind, DeclarationScope expectedScope)
+        {
+            Assert.Equal(expectedRefKind, parameter.RefKind);
+            // https://github.com/dotnet/roslyn/issues/61647: Use public API.
+            //Assert.Equal(expectedScope == DeclarationScope.RefScoped, parameter.IsRefScoped);
```

Comment:
>//Assert.Equal(expectedScope == DeclarationScope.RefScoped, parameter.IsRefScoped); [](http://example.com/codeflow?start=12&length=83)

Is it worth testing through internal API for now?

Reply:
The only caller is the preceding method which tests the internal API `ParameterSymbol.Scope`.